### PR TITLE
add geolocation

### PR DESCRIPTION
**Pull request by herimanana301**

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -20,13 +20,17 @@ function App() {
...
 
   const location = Geolocation();
-  if (location.loaded) {
-    axios
-      .get(
-        `https://api-adresse.data.gouv.fr/reverse/?lon=${location.coordinates.lng}&lat=${location.coordinates.lat}`
-      )
-      .then((response) => setCity(response.data.features["0"].properties.city));
-  }
+  React.useEffect(() => {
```
Comment (Justine-Merlin) [inline frontend/src/App.jsx:23]: You can't have multiple useEffect in one component, merge your code inside the other.

**Review by Justine-Merlin — CHANGES_REQUESTED** [review]

File: frontend/src/App.jsx (hunk 1 above)
Comment (herimanana301) [reply frontend/src/App.jsx]: I change some many thing like the navigation bar that integer a button that ask the location

File: frontend/src/components/ItemsList/StationsInfo.jsx [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
-      <img
-        className="imageFavorite"
-        src="https://via.placeholder.com/20"
-        alt="state"
-      />
+      <button type="button" onClick={handleFavorite}>
+        <img
+          className={isfavorite ? "isFavorite" : "notFavorite"}
+          src={isfavorite ? fullStar : emptyStar}
+          alt="state"
```
Comment (Justine-Merlin) [inline frontend/src/components/ItemsList/StationsInfo.jsx:18]: This alt does not reflect what the img is. Rename it for accessibility.

File: frontend/src/components/navbar/NavBar.jsx [hunk 3]
```diff
@@ -10,6 +11,9 @@ function NavBar(props) {
     <header>
       <img className="icon" src={logo} alt="The logo" />
       <Search setVille={setVille} />
+      <button className="button_filter" type="button">
+        <img src={funnel} alt="FIltre" />
```
Comment (Justine-Merlin) [inline frontend/src/components/navbar/NavBar.jsx:15]: You mean filtre for alt img ?

File: frontend/src/components/navbar/NavBar.jsx [hunk 4]
```diff
@@ -10,6 +11,9 @@ function NavBar(props) {
     <header>
       <img className="icon" src={logo} alt="The logo" />
       <Search setVille={setVille} />
+      <button className="button_filter" type="button">
```
Comment (Justine-Merlin) [inline frontend/src/components/navbar/NavBar.jsx:14]: Keep your code convention for css classes and rename it in buttonFilter, don't forget to rename it in css.

**Review by Justine-Merlin — CHANGES_REQUESTED** [review]

**Comment by Justine-Merlin** [comment]
You have conflicts to resolve, it's almost done ;)

**Pull request merged**